src/cpu/mod.rs

```diff
@@ -378,6 +378,13 @@ impl Cpu {
         }
     }
 
+    fn set_bit(&mut self, regop: Register8Bit, offset: u32, state: bool) {
```

Comment:
use `usize` instead of `u32`.  Or `u8`.  We know that this value is going to be 0-7.  Or if you feel the need to restrict it use an enum and cast it as a `usize` or `u8` when shifting.